internal/cel/library/sigstore.go

```diff
@@ -0,0 +1,564 @@
+				sigstoreKeylessPrefixVerifierType,
+				cel.FunctionBinding(sigstoreVerifierBuilderKeylessPrefix),
+			),
+			cel.MemberOverload("kw_sigstore_keyless_verifier_keyless_prefix",
+				[]*cel.Type{sigstoreKeylessPrefixVerifierType, cel.StringType, cel.StringType},
+				sigstoreKeylessPrefixVerifierType,
+				cel.FunctionBinding(sigstoreKeylessPrefixVerifierKeylessPrefix),
+			),
+		),
+		cel.Function("github",
+			cel.MemberOverload("kw_sigstore_verifier_builder_github",
+				[]*cel.Type{sigstoreVerifierBuilderType, cel.StringType, cel.StringType},
+				sigstoreGitHubVerifierType,
+				cel.FunctionBinding(sigstoreVerifierBuilderGitHub),
+			),
+		),
+		cel.Function("certificate",
+			cel.MemberOverload("kw_sigstore_verifier_builder_certificate",
+				[]*cel.Type{sigstoreVerifierBuilderType, cel.StringType},
+				sigstoreCertificateVerifierType,
+				cel.BinaryBinding(sigstoreVerifierBuilderCertificate),
+			),
+		),
+		cel.Function("certificateChain",
+			cel.MemberOverload("kw_sigstore_certificate_verifier_certificate_chain",
+				[]*cel.Type{sigstoreCertificateVerifierType, cel.StringType},
+				sigstoreCertificateVerifierType,
+				cel.BinaryBinding(sigstoreCertificateVerifierCertificateChain),
+			),
+		),
+		cel.Function("requireRekorBundle",
+			cel.MemberOverload("kw_sigstore_certificate_verifier_require_rekor_bundle",
+				[]*cel.Type{sigstoreCertificateVerifierType, cel.BoolType},
+				sigstoreCertificateVerifierType,
+				cel.BinaryBinding(sigstoreCertificateVerifierRequireRekorBundle),
+			),
+		),
+		cel.Function("verify",
+			cel.MemberOverload("kw_sigstore_pub_keys_verifier_verify",
+				[]*cel.Type{sigstorePubKeysVerifierType},
+				types.DynType,
+				cel.UnaryBinding(sigstorePubKeysVerifierVerify),
+			),
+			cel.MemberOverload("kw_sigstore_keyless_verifier_verify",
+				[]*cel.Type{sigstoreKeylessVerifierType},
+				types.DynType,
+				cel.UnaryBinding(sigstoreKeylessVerifierVerify),
+			),
+			cel.MemberOverload("kw_sigstore_keyless_prefix_verifier_verify",
+				[]*cel.Type{sigstoreKeylessPrefixVerifierType},
+				types.DynType,
+				cel.UnaryBinding(sigstoreKeylessPrefixVerifierVerify),
+			),
+			cel.MemberOverload("kw_sigstore_github_verifier_verify",
+				[]*cel.Type{sigstoreGitHubVerifierType},
+				types.DynType,
+				cel.UnaryBinding(sigstoreKeylessGitHubActionsVerifierVerify),
+			),
+			cel.MemberOverload("kw_sigstore_certificate_verifier_verify",
+				[]*cel.Type{sigstoreCertificateVerifierType},
+				types.DynType,
+				cel.UnaryBinding(sigstoreCertificateVerifierVerify),
+			),
+		),
+	}
+}
+
+func (*sigstoreLib) ProgramOptions() []cel.ProgramOption {
+	return []cel.ProgramOption{}
+}
+
+func sigstoreImage(arg ref.Val) ref.Val {
+	image, ok := arg.Value().(string)
+	if !ok {
+		return types.MaybeNoSuchOverloadErr(arg)
+	}
+
+	return sigstoreVerifierBuilder{receiverOnlyObjectVal: receiverOnlyVal(sigstoreVerifierBuilderType), image: image}
+}
+
+func sigstoreVerifierBuilderAnnotation(args ...ref.Val) ref.Val {
```

Comment:
Maybe it would be useful to add a comment that includes the syntax of the CEL program, just as a reminder/reference of what the different args are supposed to be

Reply:
@flavio I will add a doc comment similar to the one in the Kubernetes library. Is this what you are suggesting?